paddle/framework/scope.cc

```diff
@@ -65,6 +65,13 @@ void Scope::DropKids() {
   kids_.clear();
 }
 
+void Scope::DeleteScope(Scope* scope) {
+  auto it = std::find(this->kids_.begin(), this->kids_.end(), scope);
+  PADDLE_ENFORCE(it != this->kids_.end(), "Cannot find %p as kid scope", scope);
+  this->kids_.erase(it);
+  delete scope;
+}
```

Comment:
It fits our needs.
Sidenote: do we need to make it recursive?

Reply:
Just keep it simple now. Change it if we need recursive.